src/newfight.cpp

```diff
@@ -833,7 +843,17 @@ void hit_with_multiweapon_toggle(struct char_data *attacker, struct char_data *v
     // If the attack's power is greater, subtract double the level from it.
     if (IS_SPIRIT(def->ch) || IS_ELEMENTAL(def->ch)) {
       if (att->ranged->power <= GET_LEVEL(def->ch) * 2) {
-        damage(att->ch, def->ch, 0, att->ranged->dam_type, att->ranged->is_physical);
+        bool target_died = 0;
+        target_died = damage(att->ch, def->ch, 0, att->ranged->dam_type, att->ranged->is_physical);
+        
+        //Handle suprise attack/alertness here -- spirits ranged.
+        if (!target_died && IS_NPC(def->ch)) {
+          if (AFF_FLAGGED(def->ch, AFF_SURPRISE))
+            AFF_FLAGS(def->ch).RemoveBit(AFF_SURPRISE);
+           
+          GET_MOBALERT(def->ch) = MALERT_ALERT;
```

Comment:
This one does need to be MALERT_ALARM as well. MALERT_ALERT only really comes into play when they hear something fishy like gunfire. I know death screams also put them in an alert state, but I'm not sure if it's MALERT_ALERT or _ALARM. Regardless, being in active combat alarms them.

Reply:
I can also just fix this myself after merge if you don't want to switch branches again.